### ISSUE TITLE
Lack of good documentation

### ISSUE THREAD

**Issue by claudiostevanatocylix** [odoo/documentation#5772]
I am trying to follow the tutorial "Customize a view type" for Odoo version 16:
https://www.odoo.com/documentation/16.0/developer/howtos/javascript_view.html

But at the end, it leaves us with the point number 6 "Use the view in an arch." which is nothing useful.

Odoo, please, make a better documentation, there are a lot of missing parts or parts that leaves things as is, without going into the deep of it.

I would really like to create a custom view by myself, but this lack of effort from Odoo about documentation is really annoying.

**Comment by fdardenne** [comment]
Hello @claudiostevanatocylix, thanks for the feedback.
In the previous points, you learned on how to create a new view in JavaScript. In point number 6, you can use the newly created view like we do for other views. You can find how here: https://www.odoo.com/documentation/16.0/developer/tutorials/backend.html?highlight=view%20arch#basic-views 

Would you be more comfortable with a more detailed example like this ?
```xml
    <record id="my_beautiful_view" model="ir.ui.view">
        <field name="name">my_view</field>
        <field name="model">my_model</field>
        <field name="arch" type="xml">
            <beautiful fieldFromTheArch="res.partner"/>
        </field>
    </record```

The howtos on creating a view from scratch only highlight essential steps as it is an advanced topic. 
Maybe you will be more comfortable with the "Creating a view" tutorial where things are explained a little more, references are given and commit solutions are available

**Comment by claudiostevanatocylix** [comment]
@fdardenne Yeah of course I would appreciate a more detailed documentation, who does not like it?
The fact is that I have the need to be able to create my views with many customized components, a customer where I am working is planning on a lot of personalized HTML and JavaScript.

In that tutorial I also saw that a file was missing the /** @odoo-module */ tag, the render specifically.

And now that I have the JS and CSS I had to found by myself that I needed to write this in the manifest

 'assets': {
      'web.assets_backend': [
          'rg_custom_view/static/src/views/**/*',
      ],
 },

This is what I mean with lack of effort, the user (myself in this case) must try things out and hope it is the right direction.

Now that I loaded the views, I have errors in the JavaScript, I really think that when something is written out (a documentation file) there should be some properly tests done.
